### PR TITLE
Fix: site fragment to be the accept the second parameters to have dots in them

### PR DESCRIPTION
**Pull request by enejb**
Fixes https://github.com/Automattic/wp-calypso/issues/9405

By making sure that the site id gets returned by getSiteFragment as expected.
Which means 

To test
create a user with a dot in the username on your jetpack site. See https://codex.wordpress.org/Function_Reference/sanitize_user

visit http://calypso.localhost:3000/people/team/jetpacksite.com
and notice that you can click on a user that has a dot in the username.

cc @ebinnion

### CODE REVIEW

File: client/lib/route/path.js [hunk 1]
```diff
@@ -23,26 +23,32 @@ var statsLocationsByTab = {
...
-	for ( let i = 2; i > 0; i-- ) {
-		const piece = pieces[ pieces.length - i ];
-		if ( piece && -1 !== piece.indexOf( '.' ) ) {
-			return piece;
-		}
+	const last_piece = pieces[ pieces.length - 1 ];
+	if ( last_piece && -1 !== last_piece.indexOf( '.' ) ) {
+		return last_piece;
+	}
+	// People section can have usernames in the url that have dits in the name.
```
Comment (ebinnion) [inline client/lib/route/path.js]: "dits" -> "dots"

**Comment by ebinnion** [comment]
My biggest issue with this approach is that it's adding section specific logic (people) into the application-wide routing. Another approach would be to use user IDs instead of usernames, which the API already support.

It's a bit uglier in the URL perhaps, but it's also more resistant to issues like this.

**Comment by enejb** [comment]
I would keep things as they are url wise.  I do think using usernames in url make them more usable. 
User names should be unique to each site. I think it would be great if we supported more options 
like user IDs and emails in the urls.

**Comment by ebinnion** [comment]
Closing in favor of #9445.

**Pull request closed without merging**